### PR TITLE
MINIFICPP-758: Make CoAP default protocol and add properties as ident…

### PR DESCRIPTION
**Pull request by phrocker**
…ifiers

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: conf/minifi.properties [hunk 1]
```diff
@@ -37,10 +37,37 @@ nifi.database.content.repository.directory.default=${MINIFI_HOME}/content_reposi
...
 
+## Enabling C2 Uncomment each of the following options
+## define those with missing options
+#nifi.c2.enable=true
+## define protocol parameters
+#c2.agent.protocol.class=CoapProtocol
+#nifi.c2.agent.coap.host=
+#nifi.c2.agent.coap.port=
+#c2.rest.url=
+#c2.rest.url.ack=
```
Comment (apiri) [inline conf/minifi.properties]: I understand we handle both c2.* and nifi.c2.* but can we migrate these for the sake of consistency?  Should be okay from the standpoint that we shouldn't break existing configs

File: conf/minifi.properties [hunk 2]
```diff
@@ -37,10 +37,37 @@ nifi.database.content.repository.directory.default=${MINIFI_HOME}/content_reposi
 #nifi.rest.api.user.name=admin
 #nifi.rest.api.password=password
 
+## Enabling C2 Uncomment each of the following options
+## define those with missing options
+#nifi.c2.enable=true
+## define protocol parameters
+#c2.agent.protocol.class=CoapProtocol
```
Comment (apiri) [inline conf/minifi.properties]: We need to update the language surrounding this PR's changes in C2.md as well
https://github.com/apache/nifi-minifi-cpp/blob/master/C2.md#protocols

Comment (phrocker) [reply conf/minifi.properties]: Sorry had a dirty repo with other changes and didn't include this . Thanks!

File: conf/minifi.properties (hunk 1 above)
Comment (phrocker) [reply conf/minifi.properties]: Oh shoot, good point. Thanks!

**Pull request closed without merging**